src/etc/gdb_rust_pretty_printing.py

```diff
@@ -333,27 +333,37 @@ def children(self):
 # Yield each key (and optionally value) from a BoxedNode.
 def children_of_node(boxed_node, height, want_values):
     node_ptr = boxed_node['ptr']['pointer']
-    if height > 0:
-        type_name = str(node_ptr.type.target()).replace('LeafNode', 'InternalNode')
-        node_type = gdb.lookup_type(type_name)
-        node_ptr = node_ptr.cast(node_type.pointer())
-        leaf = node_ptr['data']
-    else:
-        leaf = node_ptr.dereference()
-    keys = leaf['keys']
-    if want_values:
-        values = leaf['vals']
-    length = int(leaf['len'])
-    for i in xrange(0, length + 1):
+    head = node_ptr.dereference()
+    length = int(head['len'])
+    if length > 0:
+        type_name = str(node_ptr.type.target()) # alloc::...::NodeHeader<K, V, K2>
+        assert type_name.endswith(", ()>")
+        type_name = type_name[:-5] + ">"
         if height > 0:
-            child_ptr = node_ptr['edges'][i]['value']['value']
-            for child in children_of_node(child_ptr, height - 1, want_values):
-                yield child
-        if i < length:
-            if want_values:
-                yield (keys[i]['value']['value'], values[i]['value']['value'])
-            else:
-                yield keys[i]['value']['value']
+            type_name = type_name.replace('NodeHeader', 'InternalNode', 1)
```

Comment:
The missing `,1 ` in `replace` is definitely a bug in rust-gdb nightly, but I'm not planning on making an issue for it.

Reply:
moved off to #68098